agent/consul/peering_backend.go

```diff
@@ -51,10 +53,39 @@ func (b *PeeringBackend) GetLeaderAddress() string {
 	return b.leaderAddr
 }
 
-// GetAgentCACertificates gets the server's raw CA data from its TLS Configurator.
-func (b *PeeringBackend) GetAgentCACertificates() ([]string, error) {
-	// TODO(peering): handle empty CA pems
-	return b.srv.tlsConfigurator.GRPCManualCAPems(), nil
+// GetTLSMaterials returns the TLS materials for the dialer to dial the acceptor using TLS.
+// It returns the server name to validate, and the CA certificate to validate with.
+func (b *PeeringBackend) GetTLSMaterials() (string, []string, error) {
+	// Do not send TLS materials to the dialer if we to not have TLS configured for gRPC.
+	if b.srv.config.GRPCTLSPort <= 0 && !b.srv.tlsConfigurator.GRPCServerUseTLS() {
+		return "", nil, nil
```

Comment:
This may actually need to change to always return a server name, need to think about it a bit more

Reply:
Updated to require TLS + Connect. No longer considering the manual cert in favor of auto-TLS.